cmake/FindOpenSSL.cmake

```diff
@@ -0,0 +1,28 @@
+#  OPENSSL_ROOT_DIR - Set this variable to the root installation of OpenSSL
+#
+# Read-Only variables:
+#  OPENSSL_FOUND - system has the OpenSSL library
+#  OPENSSL_INCLUDE_DIR - the OpenSSL include directory
+#  OPENSSL_LIBRARIES - The libraries needed to use OpenSSL
+#  OPENSSL_VERSION - This is set to $major.$minor.$revision$path (eg. 0.9.8s)
+
+find_path(OPENSSL_INCLUDE_DIR
+    NAMES openssl/ssl.h
```

Comment:
openssl is used pretty widely and very stable and they provide bug fix regularly to plug the security hole.